tests/testlib/s2n_connection_test_utils.c

```diff
@@ -180,8 +180,12 @@ int s2n_connections_set_io_pair(struct s2n_connection *client, struct s2n_connec
 
 int s2n_io_pair_close(struct s2n_test_io_pair *io_pair)
 {
-    POSIX_GUARD(s2n_io_pair_close_one_end(io_pair, S2N_CLIENT));
-    POSIX_GUARD(s2n_io_pair_close_one_end(io_pair, S2N_SERVER));
+    if (fcntl(io_pair->client, F_GETFD) != -1 || errno != EBADF) {
```

Comment:
I discussed this change with Lindsay, and she had the idea to set the file descriptor to -1 after we close it. That way we can check for > -1 before trying to close it again rather than checking to see if the file descriptor is valid. This will avoid cases where the previous file descriptor is re-assigned to a valid value and then unintentionally closed by `s2n_io_pair_close()`.

So could `s2n_io_pair_close_one_end()` maybe set the file descriptor to -1 after it closes it? I also wonder if the >-1 validity check should go in `s2n_io_pair_close_one_end()` rather than `s2n_io_pair_close()`, in case `s2n_io_pair_close_one_end()` is called twice as well.

Reply:
I have committed Lindsay's suggestions.

Reply:
The change is moved to [PR#4841](https://github.com/aws/s2n-tls/pull/4841).